lib/util.js

```diff
@@ -134,6 +136,203 @@ exports.nowTimestamp = function nowTimestamp() {
 	return Date.now() / 1000;
 };
 
+/**
+ * @typedef StatsEntry {}
+ * @property value {*}
+ * @property labels {object}
+ */
+
+/**
+ * Lookup table for stats by labels.
+ */
+class LabelMap {
+	/** @type {Set<string>} */
+	#labelNames;
+
+	/** @type {Map<string, StatsEntry>} */
+	#map = new Map();
+
+	constructor(labelNames = []) {
+		this.#labelNames = new Set(labelNames.slice().sort());
```

Comment:
```suggestion
		this.#labelNames = new Set(labelNames);
```

is there any reason to sort when storing in a `Set`?